From 8e7e5b841a46649afe260e50dd5d07a4fa57e4d3 Mon Sep 17 00:00:00 2001
From: allen-bst <allen@bluestacks.com>
Date: Tue, 28 Nov 2023 15:26:13 +0800
Subject: [PATCH 17/21] Revert "SW encode: Fix output buffer size"

This reverts commit 1bede7f6a837cc2d7ff1bbb86d9af096d688a3e6.
This commit causes the camera to be unable to record video.
---
 .../omx/SoftVideoEncoderOMXComponent.cpp      | 27 -------------------
 1 file changed, 27 deletions(-)

diff --git a/media/libstagefright/omx/SoftVideoEncoderOMXComponent.cpp b/media/libstagefright/omx/SoftVideoEncoderOMXComponent.cpp
index 11e563cdc2..d75acdabf8 100644
--- a/media/libstagefright/omx/SoftVideoEncoderOMXComponent.cpp
+++ b/media/libstagefright/omx/SoftVideoEncoderOMXComponent.cpp
@@ -377,33 +377,6 @@ OMX_ERRORTYPE SoftVideoEncoderOMXComponent::internalGetParameter(
             return OMX_ErrorNone;
         }
 
-        case OMX_IndexParamPortDefinition:
-        {
-            OMX_PARAM_PORTDEFINITIONTYPE *def =
-                (OMX_PARAM_PORTDEFINITIONTYPE *)param;
-
-            if (def->nPortIndex > 1) {
-                return OMX_ErrorUndefined;
-            }
-
-            OMX_ERRORTYPE err = SimpleSoftOMXComponent::internalGetParameter(index, param);
-            if (OMX_ErrorNone != err) {
-                return err;
-            }
-
-            def->format.video.nFrameWidth = mWidth;
-            def->format.video.nFrameHeight = mHeight;
-
-            // XXX: For now just configure input and output buffers the same size.
-            // May want to determine a more suitable output buffer size independent
-            // of YUV format.
-            CHECK(mColorFormat == OMX_COLOR_FormatYUV420Planar ||
-                    mColorFormat == OMX_COLOR_FormatYUV420SemiPlanar);
-            def->nBufferSize = mWidth * mHeight * 3 / 2;
-
-            return OMX_ErrorNone;
-        }
-
         default:
             return SimpleSoftOMXComponent::internalGetParameter(index, param);
     }
